lambda-layer/patches/opentelemetry-java-instrumentation.patch

```diff
@@ -636,10 +636,10 @@ index cc1414c0bf..db8a59b046 100644
 --- a/version.gradle.kts
 +++ b/version.gradle.kts
 @@ -1,5 +1,5 @@
--val stableVersion = "1.32.1"
--val alphaVersion = "1.32.1-alpha"
-+val stableVersion = "1.32.1-adot-lambda1"
-+val alphaVersion = "1.32.1-adot-lambda1-alpha"
+-val stableVersion = "1.33.6"
+-val alphaVersion = "1.33.6-alpha"
++val stableVersion = "1.33.6-adot-lambda1"
```

Comment:
In current build-layer.sh, the [lambda layer](https://github.com/aws-observability/aws-otel-java-instrumentation/blob/dc6ba3742a81f6dce2b15965a1012f42cfd9a83e/lambda-layer/build-layer.sh#L23) is applied first, then the apply the adot patch. Whereas in [adot patch](https://github.com/aws-observability/aws-otel-java-instrumentation/blob/dc6ba3742a81f6dce2b15965a1012f42cfd9a83e/.github/patches/opentelemetry-java-instrumentation.patch#L1360) the version number is changed from:
```
-val stableVersion = "1.33.6"
-val alphaVersion = "1.33.6-alpha"
```
to 
```
+val stableVersion = "1.33.6-adot1"
+val alphaVersion = "1.33.6-adot1-alpha"
```
Does this mean the lambda patch should continue use the following version numbers?
```
-val stableVersion = "1.33.6"
-val alphaVersion = "1.33.6-alpha"
```

Reply:
Good callout. As we discussed offline, I have reversed the patching order and slightly modified the lambda layer patch to not conflict (and get rejected) anymore. Much cleaner this way. :)